add openshift compatibility mode (#2358)
VictoriaMetrics/operator: docs/CHANGELOG.md

* FEATURE: [vmoperator](https://docs.victoriametrics.com/operator/): add `useLegacyNaming` field to all operator CRs. When enabled, managed resources are named using the standalone Helm chart convention (CR name directly for single-component CRs; `<name>-<component>` for cluster CRs) instead of the default operator prefix convention. Useful when migrating from standalone charts to operator-managed resources without recreating existing resources.
* FEATURE: [vmdistributed](https://docs.victoriametrics.com/operator/resources/vmdistributed/): support using `VMSingle` backends alongside `VMCluster` backends in separate zones.

> * FEATURE: [vmoperator](https://docs.victoriametrics.com/operator/): add `VM_OPENSHIFT_COMPATIBILITY` env var (`OpenshiftCompatibilityMode` config field, default `auto`) to enable OpenShift-specific compatibility features for managed pods. In `auto` mode the operator detects OpenShift at startup via the `security.openshift.io/v1` API group. Currently covers automatic mounting of the `openshift-service-ca.crt` ConfigMap into `VMAgent` and `VMSingle` pods at `/etc/ssl/certs/openshift-service-ca/service-ca.crt`, allowing scrape targets signed with the OpenShift service signing CA to be verified without per-target TLS config. Supported values: `auto`, `enabled`, `disabled`. See [#2971](https://github.com/VictoriaMetrics/helm-charts/issues/2971).

Reviewer: ```suggestion
* FEATURE: [vmoperator](https://docs.victoriametrics.com/operator/): add `VM_OPENSHIFT_COMPATIBILITY` env var (`OpenshiftCompatibilityMode` config field, default `auto`) to enable OpenShift-specific compatibility features for managed pods. In `auto` mode the operator detects OpenShift at startup via presence of `security.openshift.io/v1` API group. Currently covers automatic mounting of the `openshift-service-ca.crt` ConfigMap into `VMAgent` and `VMSingle` pods at `/etc/ssl/certs/openshift-service-ca/service-ca.crt`, so that targets signed with the OpenShift service signing CA could be verified without additional TLS config. Supported values: `auto`, `enabled`, `disabled`. See [#2971](https://github.com/VictoriaMetrics/helm-charts/issues/2971).
```
